crates/sui-network/src/state_sync/test_utils.rs

```diff
@@ -125,43 +128,57 @@ impl CommitteeFixture {
         checkpoint
     }
 
+    #[allow(clippy::type_complexity)]
     pub fn make_checkpoints(
         &self,
         number_of_checkpoints: usize,
         previous_checkpoint: Option<VerifiedCheckpoint>,
+        random_content: bool,
```

Comment:
the naked bool makes it a bit hard to tell at callsites what this argument does. maybe pass in the checkpoint contents generator function (i.e. `random_contents` or `empty_contents`) as an argument instead? that removes the extra "if" below, and also makes it clearer at call sites what this argument is for.